internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,445 @@
+		output.Println(addDryRunPrefix(uninstallStr))
+		return nil
+	}
+
+	if err := os.RemoveAll(pathToPlugin); err != nil {
+		return err
+	}
+
+	output.Println(uninstallStr)
+	return nil
+}
+
+func installFromLocal(pluginManifest *cpstructs.Manifest, archivePath, pluginDir string) error {
+	zipReader, err := zip.OpenReader(archivePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open local archive file %s", archivePath)
+	}
+	defer zipReader.Close()
+
+	return unzipPlugin(pluginManifest, zipReader.File, pluginDir)
+}
+
+func (c *pluginCommand) installFromRemote(pluginManifest *cpstructs.Manifest, pluginDir string) error {
+	archive, err := c.HubClient.GetRemoteArchive(pluginManifest)
+	if err != nil {
+		return err
+	}
+
+	checksumErrorMsg := `%s checksum for downloaded archive (%s) does not match checksum in manifest (%s) for plugin "%s"`
+	calculatedMd5Checksum := fmt.Sprintf("%x", md5.Sum(archive))
+	if calculatedMd5Checksum != pluginManifest.Archive.Md5 {
+		return errors.Errorf(checksumErrorMsg, "MD5", calculatedMd5Checksum, pluginManifest.Archive.Md5, pluginManifest.Name)
+	}
+	calculatedSha1Checksum := fmt.Sprintf("%x", sha1.Sum(archive))
+	if calculatedSha1Checksum != pluginManifest.Archive.Sha1 {
+		return errors.Errorf(checksumErrorMsg, "SHA1", calculatedSha1Checksum, pluginManifest.Archive.Sha1, pluginManifest.Name)
+	}
+
+	zipReader, err := zip.NewReader(bytes.NewReader(archive), int64(len(archive)))
+	if err != nil {
+		return errors.Wrapf(err, "failed to open remote archive file %s", archive)
+	}
+
+	return unzipPlugin(pluginManifest, zipReader.File, pluginDir)
+}
+
+func unzipPlugin(pluginManifest *cpstructs.Manifest, zipFiles []*zip.File, pluginDir string) error {
+	relativeInstallationDir := filepath.Join(pluginDir, fmt.Sprintf("%s-%s", pluginManifest.Owner.Username, pluginManifest.Name))
+	installationDir, err := filepath.Abs(relativeInstallationDir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve absolute path for directory %s", relativeInstallationDir)
+	}
+
+	for _, zipFile := range zipFiles {
+		versionPrefix := fmt.Sprintf("%s-%s-%s", pluginManifest.Owner.Username, pluginManifest.Name, pluginManifest.Version)
+		destFilePath := filepath.Join(installationDir, strings.TrimPrefix(zipFile.Name, versionPrefix))
+
+		createDirectoryErrorMsg := "failed to create directory %s on local storage"
+		if zipFile.FileInfo().IsDir() {
+			if err := os.MkdirAll(destFilePath, 0755); err != nil {
+				return errors.Wrapf(err, createDirectoryErrorMsg, destFilePath)
+			}
+			continue
+		} else {
+			if err := os.MkdirAll(filepath.Dir(destFilePath), 0755); err != nil {
+				return errors.Wrapf(err, createDirectoryErrorMsg, filepath.Dir(destFilePath))
+			}
+		}
+
+		zipFileReader, err := zipFile.Open()
+		if err != nil {
+			return errors.Wrapf(err, "failed to read file %s from archive", zipFile.Name)
+		}
+		defer zipFileReader.Close()
+
+		destFile, err := os.OpenFile(destFilePath, os.O_CREATE|os.O_RDWR, 0644)
```

Comment:
Why do we need to set the file permissions here?

Reply:
It's a required argument for this function, so I just went with Chris' value. This one gives rdwr permissions to the owner and rd permission to everyone else, which seems ok.